src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -492,196 +543,248 @@ protected void Accepting(object message)
+            });
+
+            Receive<Send>(send =>
+            {
+                var recipientAddress = send.Recipient.Path.Address;
+                Func<int?, IActorRef> createAndRegisterWritingEndpoint = refuseUid => _endpoints.RegisterWritableEndpoint(recipientAddress,
+                    CreateEndpoint(recipientAddress, send.Recipient.LocalAddressToUse,
+                        _transportMapping[send.Recipient.LocalAddressToUse], _settings, writing: true,
+                        handleOption: null, refuseUid: refuseUid), uid: null, refuseUid: refuseUid);
+
+                _endpoints.WritableEndpointWithPolicyFor(recipientAddress).Match()
+                    .With<Pass>(
+                        pass =>
                         {
-                            if(gated.TimeOfRelease.IsOverdue) createAndRegisterWritingEndpoint(null).Tell(send);
-                            else Context.System.DeadLetters.Tell(send);
+                            pass.Endpoint.Tell(send);
                         })
-                        .With<Quarantined>(quarantined =>
-                        {
+                    .With<Gated>(gated =>
+                    {
+                        if (gated.TimeOfRelease.IsOverdue) createAndRegisterWritingEndpoint(null).Tell(send);
+                        else Context.System.DeadLetters.Tell(send);
+                    })
+                    .With<Quarantined>(quarantined =>
+                    {
                             // timeOfRelease is only used for garbage collection reasons, therefore it is ignored here. We still have
                             // the Quarantined tombstone and we know what UID we don't want to accept, so use it.
-                            createAndRegisterWritingEndpoint((int)quarantined.Uid).Tell(send);
-                        })
-                        .Default(msg => createAndRegisterWritingEndpoint(null).Tell(send));
-                })
-                .With<InboundAssociation>(HandleInboundAssociation)
-                .With<EndpointWriter.StoppedReading>(endpoint => AcceptPendingReader(endpoint.Writer))
-                .With<Terminated>(terminated =>
-                {
-                    AcceptPendingReader(terminated.ActorRef);
-                    endpoints.UnregisterEndpoint(terminated.ActorRef);
-                    HandleStashedInbound(terminated.ActorRef);
-                })
-                .With<EndpointWriter.TookOver>(tookover => RemovePendingReader(tookover.Writer, tookover.ProtocolHandle))
-                .With<ReliableDeliverySupervisor.GotUid>(gotuid =>
-                {
-                    endpoints.RegisterWritableEndpointUid(Sender, gotuid.Uid);
-                    HandleStashedInbound(Sender);
-                })
-                .With<Prune>(prune => endpoints.Prune())
-                .With<ShutdownAndFlush>(shutdown =>
-                {
+                            createAndRegisterWritingEndpoint(quarantined.Uid).Tell(send);
```

Comment:
If we've quarantined the remote node and still want to talk to it, we'll allow it - but we let the `ReliableDeliverSupervisor` know the UID of the bad instance of this connection so it can refuse it if the remote node still hasn't restarted since being quarantined.